### DISCUSSION TITLE
Terminal command lines for complete installation please.

### DISCUSSION THREAD

**Discussion by artur5518** [rtl-airband/RTLSDR-Airband#307] [Q&A]
Hello all ! thanks for reading and helping me.

Since some changes occurred on a software I can´t do new installations on UBUNTU. I´m trying since some few days and without any success.

I´ve downloaded the last version of the Airband directly and tried to install older versions. I think something is happening with cmake or anything else.

I have many installations on PC´s and RPis but in these past 2 months I can´t do a new install.

I ask you please to put all command lines step by step to install into a fresh UBUNTU. 

Look that my RTLSDR is working 100%

My real problem is the RTL AIRBAND installation which is stopping in some step.

Here is the installation proccess until the error:

artur@artur-VGN-NW120AJ:~$ sudo apt-get install build-essential cmake pkg-config libmp3lame-dev libshout3-dev libconfig++-dev
Lendo listas de pacotes... Pronto
Construindo árvore de dependências       
Lendo informação de estado... Pronto
build-essential is already the newest version (12.4ubuntu1).
cmake is already the newest version (3.10.2-1ubuntu2.18.04.2).
The following additional packages will be installed:
  libconfig++9v5 libconfig-dev libconfig-doc libconfig9 libogg-dev
  libspeex-dev libssl-dev libtheora-dev libvorbis-dev
Pacotes sugeridos:
  libssl-doc
Os NOVOS pacotes a seguir serão instalados:
  libconfig++-dev libconfig++9v5 libconfig-dev libconfig-doc libconfig9
  libmp3lame-dev libogg-dev libshout3-dev libspeex-dev libssl-dev
  libtheora-dev libvorbis-dev pkg-config
0 pacotes atualizados, 13 pacotes novos instalados, 0 a serem removidos e 3 não atualizados.
É preciso baixar 2.986 kB de arquivos.
Depois desta operação, 13,2 MB adicionais de espaço em disco serão usados.
Você quer continuar? [S/n] S
Obter:1 http://br.archive.ubuntu.com/ubuntu bionic/universe amd64 libconfig++9v5 amd64 1.5-0.4 [31,6 kB]
Obter:2 http://br.archive.ubuntu.com/ubuntu bionic/universe amd64 libconfig9 amd64 1.5-0.4 [22,1 kB]
Obter:3 http://br.archive.ubuntu.com/ubuntu bionic/main amd64 pkg-config amd64 0.29.1-0ubuntu2 [45,0 kB]
Obter:4 http://br.archive.ubuntu.com/ubuntu bionic/universe amd64 libconfig-dev amd64 1.5-0.4 [51,3 kB]
Obter:5 http://br.archive.ubuntu.com/ubuntu bionic/universe amd64 libconfig++-dev amd64 1.5-0.4 [34,1 kB]
Obter:6 http://br.archive.ubuntu.com/ubuntu bionic/universe amd64 libconfig-doc all 1.5-0.4 [306 kB]
Obter:7 http://br.archive.ubuntu.com/ubuntu bionic/main amd64 libmp3lame-dev amd64 3.100-2 [156 kB]
Obter:8 http://br.archive.ubuntu.com/ubuntu bionic/main amd64 libogg-dev amd64 1.3.2-1 [156 kB]
Obter:9 http://br.archive.ubuntu.com/ubuntu bionic/main amd64 libspeex-dev amd64 1.2~rc1.2-1ubuntu2 [64,6 kB]
Obter:10 http://br.archive.ubuntu.com/ubuntu bionic-updates/main amd64 libssl-dev amd64 1.1.1-1ubuntu2.1~18.04.14 [1.568 kB]
Obter:11 http://br.archive.ubuntu.com/ubuntu bionic/main amd64 libtheora-dev amd64 1.1.1+dfsg.1-14 [186 kB]
Obter:12 http://br.archive.ubuntu.com/ubuntu bionic/main amd64 libvorbis-dev amd64 1.3.5-4.2 [321 kB]
Obter:13 http://br.archive.ubuntu.com/ubuntu bionic/main amd64 libshout3-dev amd64 2.4.1-2build1 [44,1 kB]
Baixados 2.986 kB em 1s (3.178 kB/s) 
A seleccionar pacote anteriormente não seleccionado libconfig++9v5:amd64.
(Lendo banco de dados ... 176047 ficheiros e directórios actualmente instalados.)
A preparar para desempacotar .../00-libconfig++9v5_1.5-0.4_amd64.deb ...
A descompactar libconfig++9v5:amd64 (1.5-0.4) ...
A seleccionar pacote anteriormente não seleccionado libconfig9:amd64.
A preparar para desempacotar .../01-libconfig9_1.5-0.4_amd64.deb ...
A descompactar libconfig9:amd64 (1.5-0.4) ...
A seleccionar pacote anteriormente não seleccionado pkg-config.
A preparar para desempacotar .../02-pkg-config_0.29.1-0ubuntu2_amd64.deb ...
A descompactar pkg-config (0.29.1-0ubuntu2) ...
A seleccionar pacote anteriormente não seleccionado libconfig-dev:amd64.
A preparar para desempacotar .../03-libconfig-dev_1.5-0.4_amd64.deb ...
A descompactar libconfig-dev:amd64 (1.5-0.4) ...
A seleccionar pacote anteriormente não seleccionado libconfig++-dev:amd64.
A preparar para desempacotar .../04-libconfig++-dev_1.5-0.4_amd64.deb ...
A descompactar libconfig++-dev:amd64 (1.5-0.4) ...
A seleccionar pacote anteriormente não seleccionado libconfig-doc.
A preparar para desempacotar .../05-libconfig-doc_1.5-0.4_all.deb ...
A descompactar libconfig-doc (1.5-0.4) ...
A seleccionar pacote anteriormente não seleccionado libmp3lame-dev:amd64.
A preparar para desempacotar .../06-libmp3lame-dev_3.100-2_amd64.deb ...
A descompactar libmp3lame-dev:amd64 (3.100-2) ...
A seleccionar pacote anteriormente não seleccionado libogg-dev:amd64.
A preparar para desempacotar .../07-libogg-dev_1.3.2-1_amd64.deb ...
A descompactar libogg-dev:amd64 (1.3.2-1) ...
A seleccionar pacote anteriormente não seleccionado libspeex-dev:amd64.
A preparar para desempacotar .../08-libspeex-dev_1.2~rc1.2-1ubuntu2_amd64.deb ...
A descompactar libspeex-dev:amd64 (1.2~rc1.2-1ubuntu2) ...
A seleccionar pacote anteriormente não seleccionado libssl-dev:amd64.
A preparar para desempacotar .../09-libssl-dev_1.1.1-1ubuntu2.1~18.04.14_amd64.deb ...
A descompactar libssl-dev:amd64 (1.1.1-1ubuntu2.1~18.04.14) ...
A seleccionar pacote anteriormente não seleccionado libtheora-dev:amd64.
A preparar para desempacotar .../10-libtheora-dev_1.1.1+dfsg.1-14_amd64.deb ...
A descompactar libtheora-dev:amd64 (1.1.1+dfsg.1-14) ...
A seleccionar pacote anteriormente não seleccionado libvorbis-dev:amd64.
A preparar para desempacotar .../11-libvorbis-dev_1.3.5-4.2_amd64.deb ...
A descompactar libvorbis-dev:amd64 (1.3.5-4.2) ...
A seleccionar pacote anteriormente não seleccionado libshout3-dev:amd64.
A preparar para desempacotar .../12-libshout3-dev_2.4.1-2build1_amd64.deb ...
A descompactar libshout3-dev:amd64 (2.4.1-2build1) ...
Configurando libssl-dev:amd64 (1.1.1-1ubuntu2.1~18.04.14) ...
Configurando libconfig++9v5:amd64 (1.5-0.4) ...
Configurando libconfig9:amd64 (1.5-0.4) ...
Configurando libspeex-dev:amd64 (1.2~rc1.2-1ubuntu2) ...
Configurando pkg-config (0.29.1-0ubuntu2) ...
Configurando libconfig-dev:amd64 (1.5-0.4) ...
Configurando libmp3lame-dev:amd64 (3.100-2) ...
Configurando libconfig-doc (1.5-0.4) ...
Configurando libogg-dev:amd64 (1.3.2-1) ...
Configurando libvorbis-dev:amd64 (1.3.5-4.2) ...
Configurando libtheora-dev:amd64 (1.1.1+dfsg.1-14) ...
Configurando libconfig++-dev:amd64 (1.5-0.4) ...
Configurando libshout3-dev:amd64 (2.4.1-2build1) ...
A processar 'triggers' para man-db (2.8.3-2ubuntu0.1) ...
A processar 'triggers' para install-info (6.5.0.dfsg.1-2) ...
A processar 'triggers' para libc-bin (2.27-3ubuntu1.4) ...
artur@artur-VGN-NW120AJ:~$ apt-get install librtlsdr-dev
E: Não foi possível abrir arquivo de trava /var/lib/dpkg/lock-frontend - open (13: Permissão negada)
E: Unable to acquire the dpkg frontend lock (/var/lib/dpkg/lock-frontend), are you root?
artur@artur-VGN-NW120AJ:~$ sudo apt-get install librtlsdr-dev
Lendo listas de pacotes... Pronto
Construindo árvore de dependências       
Lendo informação de estado... Pronto
Os NOVOS pacotes a seguir serão instalados:
  librtlsdr-dev
0 pacotes atualizados, 1 pacotes novos instalados, 0 a serem removidos e 3 não atualizados.
É preciso baixar 31,2 kB de arquivos.
Depois desta operação, 131 kB adicionais de espaço em disco serão usados.
Obter:1 http://br.archive.ubuntu.com/ubuntu bionic/universe amd64 librtlsdr-dev amd64 0.5.3-13 [31,2 kB]
Baixados 31,2 kB em 0s (99,3 kB/s) 
A seleccionar pacote anteriormente não seleccionado librtlsdr-dev.
(Lendo banco de dados ... 176539 ficheiros e directórios actualmente instalados.)
A preparar para desempacotar .../librtlsdr-dev_0.5.3-13_amd64.deb ...
A descompactar librtlsdr-dev (0.5.3-13) ...
Configurando librtlsdr-dev (0.5.3-13) ...
artur@artur-VGN-NW120AJ:~$ cd
artur@artur-VGN-NW120AJ:~$ wget -O RTLSDR-Airband-4.0.0.tar.gz https://github.com/szpajder/RTLSDR-Airband/archive/v4.0.0.tar.gz
--2022-01-20 15:26:59--  https://github.com/szpajder/RTLSDR-Airband/archive/v4.0.0.tar.gz
Resolvendo github.com (github.com)... 20.201.28.151
Conectando-se a github.com (github.com)|20.201.28.151|:443... conectado.
A requisição HTTP foi enviada, aguardando resposta... 302 Found
Localização: https://codeload.github.com/szpajder/RTLSDR-Airband/tar.gz/v4.0.0 [redirecionando]
--2022-01-20 15:27:00--  https://codeload.github.com/szpajder/RTLSDR-Airband/tar.gz/v4.0.0
Resolvendo codeload.github.com (codeload.github.com)... 20.201.28.149
Conectando-se a codeload.github.com (codeload.github.com)|20.201.28.149|:443... conectado.
A requisição HTTP foi enviada, aguardando resposta... 200 OK
Tamanho: não especificada [application/x-gzip]
Salvando em: “RTLSDR-Airband-4.0.0.tar.gz”

RTLSDR-Airband-4.0.0.ta     [ <=>                           ] 116,56K  --.-KB/s    em 0,07s   

2022-01-20 15:27:00 (1,57 MB/s) - “RTLSDR-Airband-4.0.0.tar.gz” salvo [119356]

artur@artur-VGN-NW120AJ:~$ tar xvfz RTLSDR-Airband-4.0.0.tar.gz
RTLSDR-Airband-4.0.0/
RTLSDR-Airband-4.0.0/.github/
RTLSDR-Airband-4.0.0/.github/ISSUE_TEMPLATE/
RTLSDR-Airband-4.0.0/.github/ISSUE_TEMPLATE/bug_report.md
RTLSDR-Airband-4.0.0/.github/ISSUE_TEMPLATE/config.yml
RTLSDR-Airband-4.0.0/.github/ISSUE_TEMPLATE/feature_request.md
RTLSDR-Airband-4.0.0/.github/workflows/
RTLSDR-Airband-4.0.0/.github/workflows/build.yml
RTLSDR-Airband-4.0.0/.gitignore
RTLSDR-Airband-4.0.0/CMakeLists.txt
RTLSDR-Airband-4.0.0/LICENSE
RTLSDR-Airband-4.0.0/NEWS.md
RTLSDR-Airband-4.0.0/README.md
RTLSDR-Airband-4.0.0/config/
RTLSDR-Airband-4.0.0/config/basic_multichannel.conf
RTLSDR-Airband-4.0.0/config/basic_scanning.conf
RTLSDR-Airband-4.0.0/config/mixers.conf
RTLSDR-Airband-4.0.0/config/two_dongles_multiple_outputs.conf
RTLSDR-Airband-4.0.0/init.d/
RTLSDR-Airband-4.0.0/init.d/rtl_airband-debian.sh
RTLSDR-Airband-4.0.0/init.d/rtl_airband-freebsd.sh
RTLSDR-Airband-4.0.0/init.d/rtl_airband-gentoo.sh
RTLSDR-Airband-4.0.0/init.d/rtl_airband.service
RTLSDR-Airband-4.0.0/src/
RTLSDR-Airband-4.0.0/src/.gitignore
RTLSDR-Airband-4.0.0/src/CMakeLists.txt
RTLSDR-Airband-4.0.0/src/CMakeModules/
RTLSDR-Airband-4.0.0/src/CMakeModules/FindBCM_VC.cmake
RTLSDR-Airband-4.0.0/src/CMakeModules/FindLame.cmake
RTLSDR-Airband-4.0.0/src/CMakeModules/FindMiriSDR.cmake
RTLSDR-Airband-4.0.0/src/CMakeModules/FindRTLSDR.cmake
RTLSDR-Airband-4.0.0/src/CMakeModules/version.cmake
RTLSDR-Airband-4.0.0/src/config.cpp
RTLSDR-Airband-4.0.0/src/config.h.in
RTLSDR-Airband-4.0.0/src/hello_fft/
RTLSDR-Airband-4.0.0/src/hello_fft/CMakeLists.txt
RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft.c
RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft.h
RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft.txt
RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft_base.c
RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft_shaders.c
RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft_trans.h
RTLSDR-Airband-4.0.0/src/hello_fft/gpu_fft_twiddles.c
RTLSDR-Airband-4.0.0/src/hello_fft/hex/
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_1024k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_128k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_16k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_1k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_2048k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_256.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_256k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_2k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_32k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_4k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_512.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_512k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_64k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_8k.hex
RTLSDR-Airband-4.0.0/src/hello_fft/hex/shader_trans.hex
RTLSDR-Airband-4.0.0/src/hello_fft/mailbox.c
RTLSDR-Airband-4.0.0/src/hello_fft/mailbox.h
RTLSDR-Airband-4.0.0/src/input-common.cpp
RTLSDR-Airband-4.0.0/src/input-common.h
RTLSDR-Airband-4.0.0/src/input-file.cpp
RTLSDR-Airband-4.0.0/src/input-file.h
RTLSDR-Airband-4.0.0/src/input-helpers.cpp
RTLSDR-Airband-4.0.0/src/input-helpers.h
RTLSDR-Airband-4.0.0/src/input-mirisdr.cpp
RTLSDR-Airband-4.0.0/src/input-mirisdr.h
RTLSDR-Airband-4.0.0/src/input-rtlsdr.cpp
RTLSDR-Airband-4.0.0/src/input-rtlsdr.h
RTLSDR-Airband-4.0.0/src/input-soapysdr.cpp
RTLSDR-Airband-4.0.0/src/input-soapysdr.h
RTLSDR-Airband-4.0.0/src/mixer.cpp
RTLSDR-Airband-4.0.0/src/output.cpp
RTLSDR-Airband-4.0.0/src/pulse.cpp
RTLSDR-Airband-4.0.0/src/rtl_airband.cpp
RTLSDR-Airband-4.0.0/src/rtl_airband.h
RTLSDR-Airband-4.0.0/src/rtl_airband_neon.s
RTLSDR-Airband-4.0.0/src/rtl_airband_vfp.s
RTLSDR-Airband-4.0.0/src/squelch.cpp
RTLSDR-Airband-4.0.0/src/squelch.h
RTLSDR-Airband-4.0.0/src/udp_stream.cpp
RTLSDR-Airband-4.0.0/src/util.cpp
artur@artur-VGN-NW120AJ:~$ mkdir build
artur@artur-VGN-NW120AJ:~$ cd build
artur@artur-VGN-NW120AJ:~/build$ cmake ../
**CMake Error: The source directory "/home/artur" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.**
artur@artur-VGN-NW120AJ:~/build$ 

Thanks a lot,

ARTUR  - PY2KJ

**Comment by charlie-foxtrot** [comment]
Not sure if you figured it out yet or not, but looking at the commands you ran, it looks like you un-tar'ed `RTLSDR-Airband-4.0.0.tar.gz` and that extracted all the files into a new directory `RTLSDR-Airband-4.0.0` but you didnt `cd` into that directory before making a `build` directory.  After you extract the tar, try `cd RTLSDR-Airband-4.0.0` before `mkdir build ; cd build ; cmake ../`

**Reply by artur5518** [reply]
Hello ! Thanks for reply but take a look what´s happening  in the bold lines and please try to help me:

artur@artur-VGN-NW120AJ:~/RTLSDR-Airband-4.0.0/build$ cmake -DPLATFORM=native
**CMake Error: The source directory "/home/artur/RTLSDR-Airband-4.0.0/build" does not appear to contain CMakeLists.txt.
Specify --help for usage, or press the help button on the CMake GUI.**
artur@artur-VGN-NW120AJ:~/RTLSDR-Airband-4.0.0/build$ 

Many thanks !

ARTUR - PY2KJ

**Reply by charlie-foxtrot** [reply]
you're missing the `../` part of the `cmake` command.

you want to create a new temporary directory that will be used for the build files:
`mkdir build`

then you want to go into that directory:
`cd build`

finally you want to run `cmake` to generate the makefile.  `cmake` requires the path to the root of the checkout, but because you have `cd`'ed into your temporary directory, the command is `cmake ../` (or in your case where you are setting the platform the full command would be:
`cmake -DPLATFORM=native ../`

**Reply by artur5518** [reply]
Hello my brother ! How are you ? Thanks for all help from you ! 

Can you please write all command lines to install in a UBUNTU PC ?

And please, I need to also the command lines for start the RTLSDR-Airband at a computer start. I mean automatically when I boot the PC.

Run it right together when the system start.

It will be easier for me when I have the command lines. I´m a starter LINUX user and I have some difficulties to understand some commands.

Many thanks again.

Best regards,

ARTUR - PY2KJ

**Reply by szpajder** [reply]
How about following instructions from the Wiki?

https://github.com/szpajder/RTLSDR-Airband/wiki/Dependencies

https://github.com/szpajder/RTLSDR-Airband/wiki/Compilation

> I´m a starter LINUX user and I have some difficulties to understand some commands.

Please be specific - what exactly is wrong or not clear? Help us make the Wiki better.

**Comment by artur5518** [comment]
Tomasz, how are you my friend ?

Yes I´ve followed the instructions from WIKI and it was going wrong. So, I asked there at the forum and someone which is helping me there. 

Anyway, I´m trying to not ask you and I´m trying to save my problem asking and using WIKI instructions.

Any other problem,  in the last instance I´ll call you back again, ok ?

I´m sorry.

Best regards and all the best to you and yours.

Sincerely,

Artur Herchcovitch – PY2KJ
